src/shmem_internal.h

```diff
@@ -394,6 +396,7 @@ typedef shmem_spinlock_t shmem_internal_mutex_t;
 #   endif /* ENABLE_PTHREAD_MUTEX */
 
 extern shmem_internal_mutex_t shmem_internal_mutex_alloc;
+extern shmem_internal_mutex_t shmem_internal_mutex_rand_r;
```

Comment:
We could put `#ifdef ENABLE_THREADS` around these declarations to force a build error if the mutexes are referenced when they are uninitialized.

Reply:
This is already inside an `ENABLE_THREADS` block, right?  The `else` right below these declarations corresponds to the `ENABLE_THREADS` check [here](https://github.com/Sandia-OpenSHMEM/SOS/pull/786/files/2fc6b70311fc4b02e34b17c16c01afc0883a66f9#diff-5e8cf268a51a6db4dc2bebef76bb50b2R344).